### PR TITLE
Fix 2404 Receive "Deployment bucket is not in the same region..." when using eu-west-1

### LINKED ISSUE #2404: Receive "Deployment bucket is not in the same region as the lambda function" when using us-east-1
# This is a Bug Report
## Description
- What went wrong?

When attempting to use a pre-existing S3 bucket in us-east-1 via the `deploymentBucket` configuration option, I receive an error response "Deployment bucket is not in the same region as the lambda function".

Debugging through the Serverless code, I observed that it is attempting to match a LocationConstraint returned from the AWS SDK to the region; however, [AWS Documentation](http://docs.aws.amazon.com/AmazonS3/latest/API/RESTBucketGETlocation.html) states: 

> When the bucket's region is US East (N. Virginia), Amazon S3 returns an empty string for the bucket's region

Due to this, it appears the check in **configureStack.js** to match the bucket's region with the specified region fails and I receive the error message.
- What did you expect should have happened?

I should have been able to use the pre-existing bucket.
- What was the config you used?

```
...
provider:
  name: aws
  runtime: nodejs4.3
  stage: dev
  region: us-east-1
  deploymentBucket: my-bucket-name
...
```
- What stacktrace or error message from your provider did you see?

Deployment bucket is not in the same region as the lambda function
## Additional Data
- **_Serverless Framework Version you're using**_: 1.0.2
- **_Operating System**_: OSX
- **_Stack Trace**_:
- **_Provider Error messages**_:

```
  Serverless Error ---------------------------------------

     Deployment bucket is not in the same region as the lambda
     function.
```

### PR DESCRIPTION
**Pull request by ssemyonov**
## What did you implement:

Closes #2404 

## How did you implement it:

Similar to original #2404 fix for us-east-1:
```if (result.LocationConstraint === 'EU') result.LocationConstraint = 'eu-west-1';```

## How can we verify it:

By deploying using pre-existing bucket in eu-west-1 region in one of the affected AWS accounts (which might be tricky to find).

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config/commands/resources
- [x] Enable ["Allow edits from maintainers"](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) for this PR
- [x] Change ready for review message below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

### CODE REVIEW

File: lib/plugins/aws/deploy/lib/configureStack.test.js [hunk 1]
```diff
@@ -110,4 +110,29 @@ describe('#configureStack', () => {
...
+  [
+    { region: 'eu-west-1', response: 'EU' },
+    { region: 'us-east-1', response: '' },
+  ].forEach((value) => {
+    it(`should handle inconsistent getBucketLocation responses for ${value.region} region`, () => {
+      const bucketName = 'com.serverless.deploys';
+
+      awsPlugin.options.region = value.region;
+
+      sinon
```
Comment (pmuens) [inline lib/plugins/aws/deploy/lib/configureStack.test.js]: This stub should be restored.

**Comment by eahefnawy** [comment]
Yeah this is hard to reproduce as it depends on what AWS **might** return. Considering the change is a simple if statement, I see no harm in merging without testing it out just like @pmuens mentioned.

**Comment by pmuens** [comment]
@ssemyonov thanks for the update! merging...

File: lib/plugins/aws/deploy/lib/configureStack.test.js (hunk 1 above)
Comment (ssemyonov) [inline lib/plugins/aws/deploy/lib/configureStack.test.js]: ```awsPlugin.provider.request.restore();``` added below

**Comment by ssemyonov** [comment]
cheers

**Comment by nick4eva** [comment]
@pmuens I assume this fix will be released in Serverless 1.6? Don't see it in the milestone 1.6.

**Comment by pmuens** [comment]
@nick4eva Yes, this one will be released in 1.6 (maybe even in 1.5.1 which is planned for the end of this week)!

**Pull request merged**